doc/sphinx/Pacemaker_Explained/node-options.rst

```diff
@@ -0,0 +1,485 @@
+       severities). Blackboxes are stored under ``/var/lib/pacemaker/blackbox``
+       by default, and their contents can be viewed using the ``qb-blackbox(8)``
+       command.
+
+       The blackbox recorder can be enabled at start using this variable, or at
+       runtime by sending a Pacemaker subsystem daemon process a ``SIGUSR1`` or
+       ``SIGTRAP`` signal, and disabled by sending ``SIGUSR2`` (see
+       ``kill(1)``). The blackbox will be written after a crash, assertion
+       failure, or ``SIGTRAP`` signal.
+
+       See :ref:`PCMK_debug <pcmk_debug>` for allowed subsystems.
+
+       Example: ``PCMK_blackbox="pacemaker-controld,pacemaker-fenced"``
+
+   * - .. _pcmk_trace_blackbox:
+
+       .. index::
+          pair:: node option; PCMK_trace_blackbox
+
+       PCMK_trace_blackbox
+     - :ref:`enumeration <enumeration>`
+     -
+     - *Advanced Use Only:* Write a blackbox whenever the message at the
+       specified function and line is logged. Multiple entries may be comma-
+       separated.
+
+       Example: ``PCMK_trace_blackbox="remote.c:144,remote.c:149"``
+
+   * - .. _pcmk_node_start_state:
+
+       .. index::
+          pair:: node option; PCMK_node_start_state
+
+       PCMK_node_start_state
+     - :ref:`enumeration <enumeration>`
+     - default
+     - By default, the local host will join the cluster in an online or standby
+       state when Pacemaker first starts depending on whether it was previously
+       put into standby mode. If this variable is set to ``standby`` or
+       ``online``, it will force the local host to join in the specified state.
+
+   * - .. _pcmk_fail_fast:
+
+       .. index::
+          pair:: node option; PCMK_fail_fast
+
+       PCMK_fail_fast
+     - :ref:`boolean <boolean>`
+     - no
+     - By default, if a Pacemaker subsystem crashes, the main ``pacemakerd``
+       process will attempt to restart it. If this variable is set to ``yes``,
+       ``pacemakerd`` will panic the local host instead.
+
+   * - .. _pcmk_panic_action:
+
+       .. index::
+          pair:: node option; PCMK_panic_action
+
+       PCMK_panic_action
+     - :ref:`enumeration <enumeration>`
+     - reboot
+     - Pacemaker will panic the local host under certain conditions. By default,
+       this means rebooting the host. This variable can change that behavior: if
+       ``crash``, trigger a kernel crash (useful if you want a kernel dump to
+       investigate); if ``sync-reboot`` or ``sync-crash``, synchronize
+       filesystems before rebooting the host or triggering a kernel crash. The
+       sync values are more likely to preserve log messages, but with the risk
+       that the host may be left active if the synchronization hangs.
+
+   * - .. _pcmk_authkey_location:
+
+       .. index::
+          pair:: node option; PCMK_authkey_location
+
+       PCMK_authkey_location
+     - :ref:`text <text>`
+     - /etc/pacemaker/authkey
```

Comment:
Default paths like this one will actually depend on what the `configure` script does to `pacemaker.in`. Not sure how else to represent it besides leaving the default unspecified.

Reply:
`doc/sphinx/conf.py.in` can define variables (`rst_prolog`)

Reply:
That's cool (I didn't know) but only one version can be hosted on the ClusterLabs website.

Reply:
Ah right, I wasn't thinking. Still worth doing for distro doc packages. We could maybe add something parenthetical like "may be different in your build". It sure would be handy for someone to be able to list all options with defaults in their build. ;)

Reply:
Makes sense in theory though I'm not sure off hand how we'll do it differently for the distro doc packages. Maybe something in autoconf. Or we can use the variables in `rst_prolog` all the time to make things simple. That's probably the way to go. In practice we'll probably be building the ClusterLabs docs on Fedora or similar for a while, so the default directories are unlikely to change between builds.

Reply:
`conf.py.in` is already configure-substituted, so `rst_prolog` just needs to use the configure variable and everything will automatically flow. By distro packages, I mean each distro can configure Pacemaker how they want, and if they ship a doc package, it will automatically show their configured value.

Reply:
It's not playing nicely for me yet. With the following substitution defined:
```
.. |CONFIGDIR| replace:: @CONFIGDIR@
```

It's giving me a literal `@CONFIGDIR` in the doc. I've tried a few different syntax arrangements. Will revisit in a bit.

Reply:
Doh, I forgot conf.py.in is not a configure .in file but handled specially in its Makefile -- should be able to add something there

Reply:
Thanks! I was able to get the substitution working that way. `Makefile.am`:
```
$(BOOKS:%=%/conf.py): conf.py.in
        $(AM_V_at)-$(MKDIR_P) "$(@:%/conf.py=%)"
        $(AM_V_GEN)sed                                                  \
                ...
                -e 's#%CONFIGDIR%#@CONFIGDIR@#g'                        \
```

`conf.py.in`:
```
rst_prolog="""
...
.. |CONFIGDIR| replace:: %CONFIGDIR%
"""
```

There is a minor annoyance. Substitution references in RST are considered inline markup and cannot be nested inside an inline literal (double backticks). So where I'm trying to say ` ``|CONFIGDIR|/pacemaker`` ` in the body of a paragraph, it will not be possible as such. There is no workaround (other than making the books themselves configure-replaceable, which isn't worth it) -- see for example https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#inline-markup

Reply:
Yeah that's not too big a deal. If it really bothers you, you could add a second replacement for the formatted subdirectory name.